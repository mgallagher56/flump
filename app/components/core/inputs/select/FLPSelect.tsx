import { type FC } from 'react';

import type { SelectProps } from '@chakra-ui/react';
import { Select } from '@chakra-ui/react';

import { css } from 'styled-system/css';

interface FLPSelectProps extends SelectProps {
  label: string;
  isLabelHidden?: boolean;
}

const FLPSelect: FC<FLPSelectProps> = ({
  flexDirection,
  icon,
  isLabelHidden,
  label,
  placeholder,
  variant,
  defaultValue,
  ...props
}) => {
  const columnStyles = css({
    display: 'flex',
    flexDirection: 'column',
    gap: props.gap || 2
  });

  const rowStyles = css({
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    gap: props.gap || 4
  });

  return (
    <div className={flexDirection === 'row' ? rowStyles : columnStyles}>
      <label htmlFor={label} hidden={isLabelHidden}>
        {label}
      </label>
      <Select id={label} {...props} />
    </div>
  );
};

export default FLPSelect;
