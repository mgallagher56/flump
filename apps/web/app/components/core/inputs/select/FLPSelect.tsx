import { type FC } from 'react';

import { SelectRoot, type SelectRootProps } from '@chakra-ui/react';
import { SelectContent, SelectItem, SelectLabel, SelectTrigger, SelectValueText } from '~/components/ui/select';

import { css } from 'styled-system/css';

interface FLPSelectProps extends SelectRootProps {
  label: string;
  isLabelHidden?: boolean;
  portalRef?: React.RefObject<HTMLDivElement>;
}

const FLPSelect: FC<FLPSelectProps> = ({
  collection,
  flexDirection,
  gap,
  isLabelHidden,
  label,
  portalRef,
  value,
  onValueChange
}) => {
  const columnStyles = css({
    display: 'flex',
    flexDirection: 'column',
    gap: gap || 2
  });

  const rowStyles = css({
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    gap: gap || 4
  });

  return (
    <div className={flexDirection === 'row' ? rowStyles : columnStyles}>
      <SelectRoot collection={collection} size="sm" value={value} width="320px" onValueChange={onValueChange}>
        <SelectLabel hidden={isLabelHidden} htmlFor={label}>
          {label}
        </SelectLabel>
        <SelectTrigger>
          <SelectValueText />
        </SelectTrigger>
        <SelectContent portalRef={portalRef}>
          {collection.items.map(({ id, name }) => (
            <SelectItem key={id} item={name}>
              {name}
            </SelectItem>
          ))}
        </SelectContent>
      </SelectRoot>
    </div>
  );
};

export default FLPSelect;
