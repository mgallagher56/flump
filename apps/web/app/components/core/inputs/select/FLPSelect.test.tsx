import { render } from '@testing-library/react';
import { vi } from 'vitest';

import FLPSelect from './FLPSelect';

describe('<FLPSelect />', () => {
  test('should render as expected', () => {
    const { baseElement } = render(
      <FLPSelect label="label" value={1} onChange={vi.fn()}>
        <option value="1">1</option>
        <option value="2">2</option>
        <option value="3">3</option>
      </FLPSelect>
    );
    expect(baseElement).toMatchSnapshot();
  });
  test('should render as expected as row', () => {
    const { baseElement } = render(
      <FLPSelect flexDirection="row" label="label" value={1} onChange={vi.fn()}>
        <option value="1">1</option>
        <option value="2">2</option>
        <option value="3">3</option>
      </FLPSelect>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
