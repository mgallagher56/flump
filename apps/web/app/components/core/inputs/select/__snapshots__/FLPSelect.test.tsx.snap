// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<FLPSelect /> > should render as expected 1`] = `
<body>
  <div>
    <div
      class="d_flex flex_column gap_2"
    >
      <label
        for="label"
      >
        label
      </label>
      <div
        class="chakra-select__wrapper css-42b2qy"
      >
        <select
          class="chakra-select css-1ik61og"
          id="label"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
        </select>
        <div
          class="chakra-select__icon-wrapper css-162mkon"
        >
          <svg
            aria-hidden="true"
            class="chakra-select__icon"
            focusable="false"
            role="presentation"
            style="width: 1em; height: 1em; color: currentcolor;"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<FLPSelect /> > should render as expected as row 1`] = `
<body>
  <div>
    <div
      class="d_flex flex_row items_center gap_4"
    >
      <label
        for="label"
      >
        label
      </label>
      <div
        class="chakra-select__wrapper css-42b2qy"
      >
        <select
          class="chakra-select css-1ik61og"
          id="label"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
        </select>
        <div
          class="chakra-select__icon-wrapper css-162mkon"
        >
          <svg
            aria-hidden="true"
            class="chakra-select__icon"
            focusable="false"
            role="presentation"
            style="width: 1em; height: 1em; color: currentcolor;"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;
